src/main/java/decodex/modules/ModuleManager.java

```diff
@@ -1,51 +1,142 @@
      *
      * @param moduleName Name of selected module.
+     * @param parameters Parameters for specific commands.
      * @return Selected module.
-     * @throws UnknownModuleException Not a module name of an available module.
+     * @throws UnknownModuleException If provided module name is not an available module.
+     * @throws ModuleException If module parameters are invalid.
      */
-    public Module selectModule(String moduleName) throws UnknownModuleException {
-        for (Module module : modules) {
-            if (module.getName().equals(moduleName)) {
-                logger.fine(String.format("Selected %s", module.getName()));
-                return module;
-            }
+    public Module selectModule(String moduleName, String[] parameters) throws UnknownModuleException, ModuleException {
+        Module module;
+
+        switch (moduleName) {
+        case Base64Encoder.MODULE_NAME:
+            module = prepareBase64Encoder();
+            break;
+        case Base64Decoder.MODULE_NAME:
+            module = prepareBase64Decoder();
+            break;
+        case HexEncoder.MODULE_NAME:
+            module = prepareHexEncoder();
+            break;
+        case HexDecoder.MODULE_NAME:
+            module = prepareHexDecoder();
+            break;
+        case RotEncoder.MODULE_NAME:
+            module = prepareRotEncoder(parameters);
+            break;
+        default:
+            throw new UnknownModuleException(moduleName);
         }
 
-        throw new UnknownModuleException(moduleName);
+        logger.fine(String.format("Selected %s", module.getName()));
+        return module;
+    }
+
+    /**
+     * Prepares and returns the Base64Encoder.
+     *
+     * @return The Base64Encoder object.
+     */
+    private Module prepareBase64Encoder() {
+        return new Base64Encoder();
+    }
+
+    /**
+     * Prepares and returns the Base64Decoder.
+     *
+     * @return The Base64Decoder object.
+     */
+    private Module prepareBase64Decoder() {
+        return new Base64Decoder();
+    }
+
+    /**
+     * Prepares and returns the HexEncoder.
+     *
+     * @return The HexEncoder object.
+     */
+    private Module prepareHexEncoder() {
+        return new HexEncoder();
+    }
+
+    /**
+     * Prepares and returns the HexDecoder.
+     *
+     * @return The HexDecoder object.
+     */
+    private Module prepareHexDecoder() {
+        return new HexDecoder();
+    }
+
+    /**
+     * Prepares and returns the RotEncoder with specified rotation.
+     *
+     * @param parameters String array of module parameters.
+     * @return The RotEncoder object.
+     * @throws ModuleException If the number of parameters are incorrect or invalid.
+     */
+    private Module prepareRotEncoder(String[] parameters) throws ModuleException {
+        if (parameters == null) {
+            throw new ModuleException(ErrorMessages.MISSING_MODULE_PARAMETERS);
+        }
```

Comment:
Perhaps change this to an assert? Since we are mostly using asserts for null checks of the method arguments.